contracts/examples/Collateralized.sol

```diff
@@ -149,6 +149,9 @@ contract Collateralized is TermsContract {
             getValueRepaidToDate(agreementID)
         );
 
+        // mark collateral as withdrawn.
```

Comment:
Your point about reentrancy is understood. Key thing for me to remember is no state change occurs if the function ultimately reverts, so it's fine that this write occurs before the actually transfer of value even in the case where the transfer of value fails.